plugins/org.python.pydev/tests_analysis/com/python/pydev/analysis/refactoring/quick_fixes/AddTokenAndImportStatementTest.java

```diff
@@ -0,0 +1,235 @@
+
+    public String checkImport(String test_document, String test_realImportRep, int test_maxCols) throws Exception {
+
+        Document document = new Document(test_document);
+
+        char trigger = '\n';
+        int offset = 0;
+        boolean addLocalImport = false;
+        boolean addLocalImportsOnTopOfMethod = false;
+        boolean groupImports = true;
+        int maxCols = test_maxCols;
+
+        AddTokenAndImportStatement stmt = new AddTokenAndImportStatement(document, trigger, offset, addLocalImport,
+                addLocalImportsOnTopOfMethod,
+                groupImports, maxCols);
+
+        String realImportRep = test_realImportRep;
+        int fReplacementOffset = document.getLineLength(0);
+        int fLen = 0;
+        String indentString = "";
+        String fReplacementString = "";
+        boolean appliedWithTrigger = false;
+        int importLen = 0;
+
+        ComputedInfo computedInfo = new ComputedInfo(realImportRep, fReplacementOffset, fLen, indentString,
+                fReplacementString, appliedWithTrigger, importLen, document);
+        stmt.createTextEdit(computedInfo);
+
+        for (ReplaceEdit edit : computedInfo.replaceEdit) {
+            try {
+                edit.apply(document);
+            } catch (Exception e) {
+                Log.log(e);
+            }
+        }
+
+        return document.get();
+
+    }
+
+    // normal test
+    public void test1() throws Exception {
+        String test_document = "from math import ceil";
+        String test_realImportRep = "from math import sqrt";
+        String doc = checkImport(test_document, test_realImportRep, 80);
+
+        String expected = "from math import ceil, sqrt";
+        String actual = doc;
+        assertEquals(expected, actual);
+    }
+
+    // test with comma
+    public void test2() throws Exception {
+        String test_document = "from math import ceil,";
+        String test_realImportRep = "from math import sqrt";
+        String doc = checkImport(test_document, test_realImportRep, 80);
+
+        String expected = "from math import ceil, sqrt";
+        String actual = doc;
+        assertEquals(expected, actual);
+    }
+
+    // test with comment
+    public void test3() throws Exception {
+        String test_document = "from math import ceil #foo";
+        String test_realImportRep = "from math import sqrt";
+        String doc = checkImport(test_document, test_realImportRep, 80);
+
+        String expected = "from math import ceil #foofrom math import sqrt";
+        String actual = doc;
+        assertEquals(expected, actual);
+    }
+
+    // test with parenthesis and comment
+    public void test4() throws Exception {
+        String test_document = "from math import (ceil) #foo";
+        String test_realImportRep = "from math import sqrt";
+        String doc = checkImport(test_document, test_realImportRep, 80);
+
+        String expected = "from math import (ceil) #foofrom math import sqrt";
```

Comment:
This expected is wrong, so, please fix the function so that the final result of this test is `from math import (ceil, sqrt) #foo` -- the import is being added to the comment, which isn't correct and this is a bug in the implementation (same for the other tests with comments).